.github/workflows/linux_conditional_compilation.yml

```diff
@@ -76,44 +82,267 @@ jobs:
         run: |
           bash ${OPENVINO_REPO}/install_build_dependencies.sh
 
+          # use clang as a default compiler
+          apt --assume-yes install clang
+          update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100
+          update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Setup pip cache dir
+        run: |
+          PIP_VER=$(python3 -c "import pip; print(pip.__version__)")
+          echo "Using pip version: ${PIP_VER}"
+          echo "PIP_CACHE_DIR=${PIP_CACHE_PATH}/${PIP_VER}" >> $GITHUB_ENV
+
+      - name: Install python dependencies
+        run: |
+          # For running ONNX frontend unit tests
+          python3 -m pip install --force-reinstall -r ${OPENVINO_REPO}/src/frontends/onnx/tests/requirements.txt
+
+          # For running TensorFlow frontend unit tests
+          python3 -m pip install -r ${OPENVINO_REPO}/src/frontends/tensorflow/tests/requirements.txt
+
+          # For running TensorFlow Lite frontend unit tests
+          python3 -m pip install -r ${OPENVINO_REPO}/src/frontends/tensorflow_lite/tests/requirements.txt
+
+          # For running Paddle frontend unit tests
+          python3 -m pip install -r ${OPENVINO_REPO}/src/frontends/paddle/tests/requirements.txt
+
       #
       # Build
       #
 
-      - name: CMake configure CC COLLECT
+      - name: CMake configure - CC COLLECT
         run: |
           cmake \
             -G "${{ env.CMAKE_GENERATOR }}" \
+            -DBUILD_SHARED_LIBS=OFF \
+            -DENABLE_TESTS=ON \
             -DENABLE_CPPLINT=OFF \
-            -DENABLE_GAPI_PREPROCESSING=OFF \
-            -DCMAKE_VERBOSE_MAKEFILE=ON \
+            -DENABLE_NCC_STYLE=OFF \
+            -DENABLE_INTEL_GNA=OFF \
             -DCMAKE_COMPILE_WARNING_AS_ERROR=OFF \
-            -DENABLE_FASTER_BUILD=ON \
             -DENABLE_PROFILING_ITT=ON \
             -DSELECTIVE_BUILD=COLLECT \
             -S ${OPENVINO_REPO} \
             -B ${BUILD_DIR}
 
-      - name: Build CC COLLECT
-        run: cmake --build ${BUILD_DIR} --parallel --config ${{ env.CMAKE_BUILD_TYPE }} --target openvino_intel_cpu_plugin openvino_ir_frontend benchmark_app sea_itt_lib
+      - name: Cmake build - CC COLLECT
+        run: |
+          cmake --build ${BUILD_DIR} --parallel 8 --config ${{ env.CMAKE_BUILD_TYPE }}
```

Comment:
Is there a need for an explicit `8`?

Reply:
it's a temp solution to limit concurrency of link state - during static build linkage of each final application consumes a lot of memory..